### PR TITLE
add initializerconfiguration to kubectl valid resource type

### PR DESCRIPTION
**Pull request by dixudx**
xref [kubernetes #51089](https://github.com/kubernetes/kubernetes/pull/51089): add "initializerconfiguration" as kubectl valid resource

/cc @mengqiy

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5260)

### CODE REVIEW

**Comment by mengqiy** [comment]
LGTM
@dixudx Can you please verify if any resource is missing comparing to `pkg/kubectl/cmd/cmd.go`?

**Comment by dixudx** [comment]
@mengqiy By comparing to `pkg/kubectl/cmd/cmd.go`, I found,

* the doc included a resource type named `apiservices`. Do we have such resource type? @liggitt Need your confirm. If so, we may need to add new printer for `apiservers`.
* We've got `all` in `pkg/kubectl/cmd/cmd.go` as valid resource types, which is missing in the doc.

WDYT?

File: docs/user-guide/kubectl-overview.md [hunk 1]
```diff
@@ -102,6 +102,7 @@ Resource type    | Abbreviated alias
 `events` |`ev`
 `horizontalpodautoscalers` |`hpa`
 `ingresses` |`ing`
+`initializerconfiguration` |`ic`
```
Comment (liggitt) [inline docs/user-guide/kubectl-overview.md]: We should stop adding hard coded resources like this and work off of discovery

Comment (dixudx) [reply docs/user-guide/kubectl-overview.md]: @liggitt Does `apiserver` support resources discovery now.

Comment (liggitt) [reply docs/user-guide/kubectl-overview.md]: Yes. That is what kubectl actually uses now for shortname expansion

Comment (dixudx) [reply docs/user-guide/kubectl-overview.md]: So we could remove such hard-coded [`validResources`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/cmd.go#L202) by replacing with resource discovery.

> // TODO: This should be populated using the discovery information from apiserver.

But I found `validResources` did get referenced by some static variables, such as [cmd/annotate.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/annotate.go#L75), which will be impossible to decouple.

To make it easy, could we just replace the value `validResources` with a url, like <https://github.com/kubernetes/kubernetes.github.io/blob/master/docs/user-guide/kubectl-overview.md#resource-types>.

Comment (liggitt) [reply docs/user-guide/kubectl-overview.md]: We should not reference static docs, but add a command to list valid resources and short names for the current server instead (e.g. https://github.com/kubernetes/kubernetes/pull/42873) and reference that

Comment (dixudx) [reply docs/user-guide/kubectl-overview.md]: Yes.

But for [`cmd/annotate.go`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/annotate.go#L75), we could not make the request to fetch valid resources. Seems we can only modify the current long description.

Comment (liggitt) [reply docs/user-guide/kubectl-overview.md]: The long description should reference the command to run to obtain valid resources for the current live server

Comment (dixudx) [reply docs/user-guide/kubectl-overview.md]: Thanks @liggitt. Make sense. I will work this. 

But this PR is just for the documentation. We could update it this way, right?

Comment (liggitt) [reply docs/user-guide/kubectl-overview.md]: This documentation will *always* be wrong if we continue doing it this way.

**Comment by steveperry-53** [comment]
@dixudx @liggitt What would you like to do with this doc? Publish as is, or alter according to concerns raised by @liggitt?

**Comment by liggitt** [comment]
I would not put effort into keeping this manual list of resources/shortnames updated.

**Comment by steveperry-53** [comment]
@liggitt @dixudx I like the idea of not showing lists that are likely to become out of date.
I would suggest eliminating both the Operations list and the Resource types list.

For operations, we could just link to /docs/user-guide/kubectl/{{page.version}}/.

For resource types, we could show an example command. I don't understand what that command is, but something like this:

To see a list of resources, enter this command:

    kubectl ...

The output shows ...

   ...

**Comment by dixudx** [comment]
@steveperry-53 Just keep it open for now. I will update this when related PRs, like kubernetes/kubernetes#42873, get merged.

**Comment by tengqm** [comment]
Closing this because the code side change has been abandoned.

**Pull request closed without merging**